sdk/python/feast/client.py

```diff
@@ -834,50 +864,25 @@ def ingest(
 
 
 def _build_feature_references(
-    feature_refs: List[str], default_project: str = None
+    feature_ref_strs: List[str], project: Optional[str] = None
 ) -> List[FeatureReference]:
     """
-    Builds a list of FeatureSet objects from feature set ids in order to
-    retrieve feature data from Feast Serving
+    Builds a list of FeatureReference protos from string feature set references
 
     Args:
-        feature_refs: List of feature reference strings
-            ("project/feature")
-        default_project: This project will be used if the project name is
-            not provided in the feature reference
-    """
-
-    features = []
-
-    for feature_ref in feature_refs:
-        project_split = feature_ref.split("/")
-
-        if len(project_split) == 2:
-            project, name = project_split
-        elif len(project_split) == 1:
-            name = project_split[0]
-            if default_project is None:
-                raise ValueError(
-                    f"No project specified in {feature_ref} and no default project provided"
-                )
-            project = default_project
-        else:
-            raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature"'
-            )
-
-        if len(project) == 0 or len(name) == 0:
-            raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature"'
-            )
-
-        if ":" in name:
-            raise ValueError(
-                f'Could not parse feature ref {feature_ref}, expecting "project/feature". Versions were deprecated in v0.5.0.'
-            )
+        feature_ref_strs: List of string feature references
+        project: Optionally specifies the project in the parsed feature references.
 
-        features.append(FeatureReference(project=project, name=name))
-    return features
+    Returns:
+        A list of FeatureReference protos parsed from args.
+    """
+    feature_refs = [FeatureRef.from_str(ref_str) for ref_str in feature_ref_strs]
+    ref_protos = [ref.to_proto() for ref in feature_refs]
```

Comment:
I'd prefer `feature_ref_proto` or `feature_ref_spec` just to disambiguate with `feature_set_ref`

Reply:
Corrected.